### PR TITLE
2983 lakectl ingest stages more objects than source s3 bucket

### PR DESCRIPTION
**Pull request by itai-david**
A bug fix, in the way lakectl's S3 store used pagination during walk + a test

### CODE REVIEW

File: cmd/lakectl/cmd/store/s3.go [hunk 1]
```diff
@@ -27,6 +28,10 @@ type S3Walker struct {
 	s3 s3iface.S3API
 }
 
+func (s *S3Walker) SetS3(s3 s3iface.S3API) {
```
Comment (nopcoder) [inline cmd/lakectl/cmd/store/s3.go:31]: Would prefer to expose the member, so I can initial the structure set/get this member, vs just enable a setter func.
But for this test, we don't need to expose a setter.  Calling `store.Walk` function will get us this walker and better code cover.

File: cmd/lakectl/cmd/store/factory.go [hunk 2]
```diff
@@ -45,7 +47,7 @@ func Walk(ctx context.Context, storageURI string, walkFn func(e ObjectStoreEntry
 	}
 	switch uri.Scheme {
 	case "s3":
-		svc, err := GetS3Client()
+		svc, err := GetS3Client(*aws.NewConfig())
```
Comment (nopcoder) [inline cmd/lakectl/cmd/store/factory.go:50]: Just question - what is the difference between the default we used to have and passing the default new configuration?

File: nessie/store_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	"context"
+	"net/url"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/cmd/lakectl/cmd/store"
+)
+
+const IngestTestBacketPath = "s3://nessie-system-testing-data/ingest-test-data/"
```
Comment (nopcoder) [inline nessie/store_test.go:13]: Bucket

File: nessie/store_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+func TestS3Walk(t *testing.T) {
```
Comment (nopcoder) [inline nessie/store_test.go:15]: Using the Walk function, we can test all the storage we support - not saying we need to add configuration currently to Nessie now. But writing the test function using table-driven style, we add just pass different bucket Path
Can drop the S3 from the name and use table-driven code, so adding another address we can check more storage systems.

Something like:
```go
func TestWalk(t *testing.T) {
	// Test bucket was uploaded with 2100 as the test is written. If the test fails on this number,
	// make sure there were no changes made to the bucket, or update this number accordingly
	const expectedNumObjs = 2100
	tests := []struct {
		Name       string
		BucketPath string
	}{
		{Name: "s3", BucketPath: "s3://nessie-system-testing-data/ingest-test-data/"},
	}

	for _, tt := range tests {
		t.Run(tt.Name, func(t *testing.T) {
			uri, err := url.Parse(tt.BucketPath)
			require.NoError(t, err, "URL parsing failed", tt.BucketPath)
			numObjs := 0
			store.Walk(context.Background(), uri, func(store.ObjectStoreEntry) error {
				numObjs++
				return nil
			})
			require.Equal(t, expectedNumObjs, numObjs, "Wrong number of objects detected", tt.Name)
		})
	}
}

```

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: cmd/lakectl/cmd/store/factory.go (hunk 2 above)
Comment (itai-david) [reply cmd/lakectl/cmd/store/factory.go:50]: No difference in this context, which is exactly what I needed                                                                       .
On the other hand, from the test, I can force certain params, such as the region, which is required in order to access the `nessie-system-testing-data` bucket

File: nessie/store_test.go (hunk 3 above)
Comment (itai-david) [reply nessie/store_test.go]: Done

File: nessie/store_test.go (hunk 4 above)
Comment (itai-david) [reply nessie/store_test.go:15]: I can open an issue to address this in the future - WDYT?

**Review by nopcoder — APPROVED** [review]

File: nessie/store_test.go (hunk 4 above)
Comment (itai-david) [reply nessie/store_test.go:14]: Opened #3101 and #3102

Comment (nopcoder) [reply nessie/store_test.go:14]: Think you can close/delete 3102, we would like to run all the tests and locally with the suggested code we can run specific storage.

**Review by nopcoder — APPROVED** [review]
LGTM - small comment about the issue and the test code

**Pull request merged**